doc/mid-based-signalling.md

```diff
@@ -0,0 +1,252 @@
+
+The same is sent either to a focus or a peer client.
+
+A focus will gather the various streams that it's receiving by the `conf_id` from the `m.negotiate` section. In future
+when stream are advertised via the room state, this could be unnecessary and foci need not have any knowledge of what
+group calls are happening (assuming we don't need them to enforce viewship based on this).
+
+The peer or focus then answers, again over to-device message. A peer will start sending media tracks automatically and
+therefore describe them in the answer:
+
+m.call.answer (full mesh)
+```
+{
+    "m.negotiate": {
+        "version": 1,
+        "call_id": "35657a5b793ce",
+        "conf_id": "bbe53499f82e3",
+        "party_id": "678910",
+        "description": {
+            "type": "answer",
+            "sdp": "[...]",
+        }
+    },
+    "m.call.capabilities": {
+        "m.call.transferee": false,
+        "m.call.dtmf": false,
+    },
+    "m.tracks.describe": {
+        "1": { // transceiver mid 1
+            "media_uuid": "cccc-cccc-cccccc-cccc-cccc",
+            "media_group_uuid": "2345-2345-234567-2345-2345",
+            "purpose": "m.usermedia",
+            "kind": "video"
+        },
+        "2": {
+            "media_uuid": "bbbb-bbbb-bbbbbb-bbbb-bbbb",
+            "media_group_uuid": "1234-1234-123456-1234-1234",
+            "purpose": "m.usermedia",
+            "kind": "audio",
+        },
+    },
+}
+```
+
+A focus, however, will not send any tracks by default and therefore does not include an
+`m.tracks.describe` bblock. Instead, it includes an `m.track.advertise` block advertising
+what tracks are available for that `conf_id`.
+
+m.call.answer (focus)
+```
+{
+    "m.negotiate": {
+        "version": 1,
+        "call_id": "35657a5b793ce",
+        "conf_id": "bbe53499f82e3",
+        "party_id": "678910",
+        "description": {
+            "type": "answer",
+            "sdp": "[...]",
+        }
+    },
+    "m.call.capabilities": {
+        "m.call.transferee": false,
+        "m.call.dtmf": false,
+    },
+    "m.tracks.advertise": {
+        "alice:example.org": { // user ID
+            "88888888": { // device ID
+                "2345-2345-234567-2345-2345": [{ // media group uuid
+                    "media_uuid": "aaaa-aaaa-aaaaaa-aaaa-aaaa":
+                    "purpose": "m.usermedia",
+                    "kind": "video",
+                }, {
+                    "media_uuid": "bbbb-bbbb-bbbbbb-bbbb-bbbb":
+                    "purpose": "m.usermedia",
+                    "kind": "audio",
+                },
+            },
+        }
+    },
+}
+```
+
+XXX: How flat vs deep do we want the structure to be here? I've done it quite deep here,
+organised by user ID / device ID / media group UUID, but they could also just be a flat
+list of tracks. It would be more duplication but maybe less effort to read.
```

Comment:
It would also be useful to think about how would we actually implement this and how it fits in with the current implementation

Reply:
I've expanded on this a bit in 457578a, was this the sort of thing you were thinking about?